source/fem-dem/gls_vans.cc

```diff
@@ -844,6 +847,189 @@ GLSVANSSolver<dim>::quadrature_centered_sphere_method()
+    this->cfd_dem_simulation_parameters.void_fraction
+      ->particle_refinement_factor);
+
+  DoFHandler<dim> dof_handler_particle(particle_triangulation);
+
+  FEValues<dim> fe_values_particle(mapping,
+                                   fe_particle,
+                                   quadrature_particle,
+                                   update_JxW_values |
+                                     update_quadrature_points);
+
+  dof_handler_particle.distribute_dofs(fe_particle);
+
+  std::vector<Point<dim>> reference_quadrature_location(
+    quadrature_particle.size() *
+    particle_triangulation.n_global_active_cells());
+
+  std::vector<double> reference_quadrature_weights(
+    quadrature_particle.size() *
+    particle_triangulation.n_global_active_cells());
+
+  std::vector<Point<dim>> quadrature_particle_location(
+    quadrature_particle.size() *
+    particle_triangulation.n_global_active_cells());
+
+  std::vector<double> quadrature_particle_weights(
+    quadrature_particle.size() *
+    particle_triangulation.n_global_active_cells());
+
+  double k = 0;
+  for (const auto &particle_cell : dof_handler_particle.active_cell_iterators())
+    {
+      fe_values_particle.reinit(particle_cell);
+      for (unsigned int q = 0; q < quadrature_particle.size(); ++q)
+        {
+          reference_quadrature_weights[k] =
+            (M_PI * pow(2.0, dim) / (2.0 * dim)) /
+            reference_quadrature_weights.size();
+          reference_quadrature_location[k] =
+            fe_values_particle.quadrature_point(q);
+          k += 1;
+        }
+    }
+  //-------------------------------------------------------------------------
+  for (const auto &cell :
+       this->void_fraction_dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned())
+        {
+          fe_values_void_fraction.reinit(cell);
+
+          local_matrix_void_fraction = 0;
+          local_rhs_void_fraction    = 0;
+
+          double solid_volume_in_cell = 0;
+
+          // Active Neighbors include the current cell as well
+          auto active_neighbors =
+            LetheGridTools::find_cells_around_cell<dim>(vertices_to_cell, cell);
+
+          for (unsigned int m = 0; m < active_neighbors.size(); m++)
+            {
+              // Loop over particles in cell
+              // Begin and end iterator for particles in cell
+              const auto pic =
+                particle_handler.particles_in_cell(active_neighbors[m]);
+              for (auto &particle : pic)
+                {
+                  /*****************************************************************/
+                  auto   particle_properties = particle.get_properties();
+                  auto   particle_location   = particle.get_location();
+                  double translational_factor =
+                    particle_properties[DEM::PropertiesIndex::dp] * 0.5;
+
+                  // Resize and translate reference sphere
```

Comment:
I would give  abit more detail on the calculation procedure here, this is a bit unclear

Reply:
done